### PR TITLE
Modifies workflow scripts to generate random seed when seed=-1

### PR DESCRIPTION
**Pull request by kellyguo11**
# Description

This change adds support to the train.py workflow scripts to support setting seed=-1 to generate a random seed. Previously, setting seed to -1 would cause errors to be thrown from RL libraries due to negative seed values.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def __init__(self, cfg: DirectRLEnvCfg, render_mode: str | None = None, **kwargs
 
         # set the seed for the environment
         if self.cfg.seed is not None:
-            self.seed(self.cfg.seed)
+            self.cfg.seed = self.seed(self.cfg.seed)
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py:89]: Isn't the function from Isaac Sim (torch_utils.set_seed) doing this automatically when the seed is -1?

Comment (kellyguo11) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py:89]: it will generate a random seed if seed=-1 and return the new seed, this assigns the new seed to the env cfg

File: source/standalone/workflows/skrl/train.py [hunk 2]
```diff
@@ -119,9 +120,14 @@ def main(env_cfg: ManagerBasedRLEnvCfg | DirectRLEnvCfg | DirectMARLEnvCfg, agen
     if args_cli.ml_framework.startswith("jax"):
         skrl.config.jax.backend = "jax" if args_cli.ml_framework == "jax" else "numpy"
 
+    # process seed from command line
+    if args_cli.seed == -1:
+        args_cli.seed = np.random.randint(0, 10000)
+
     # set the environment seed
```
Comment (Mayankm96) [inline source/standalone/workflows/skrl/train.py]: please fix the docstrings too accordingly

File: source/standalone/workflows/skrl/train.py [hunk 3]
```diff
@@ -119,9 +120,14 @@ def main(env_cfg: ManagerBasedRLEnvCfg | DirectRLEnvCfg | DirectMARLEnvCfg, agen
     if args_cli.ml_framework.startswith("jax"):
         skrl.config.jax.backend = "jax" if args_cli.ml_framework == "jax" else "numpy"
 
+    # process seed from command line
+    if args_cli.seed == -1:
+        args_cli.seed = np.random.randint(0, 10000)
```
Comment (Mayankm96) [inline source/standalone/workflows/skrl/train.py]: would recommend using python standard library instead of numpy for this: 

https://docs.python.org/3/library/random.html

File: source/standalone/workflows/rsl_rl/cli_args.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def update_rsl_rl_cfg(agent_cfg: RslRlOnPolicyRunnerCfg, args_cli: argparse.Name
         The updated configuration for RSL-RL agent based on inputs.
     """
     # override the default configuration with CLI arguments
-    if hasattr("args_cli", "seed") and args_cli.seed is not None:
+    if hasattr(args_cli, "seed") and args_cli.seed is not None:
```
Comment (Mayankm96) [inline source/standalone/workflows/rsl_rl/cli_args.py:71]: also needs to be propogated here for the -1 case?

File: source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py (hunk 1 above)
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py:89]: got it.

File: source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_env.py [hunk 5]
```diff
@@ -76,7 +76,7 @@ def __init__(self, cfg: ManagerBasedEnvCfg):
 
         # set the seed for the environment
         if self.cfg.seed is not None:
-            self.seed(self.cfg.seed)
+            self.cfg.seed = self.seed(self.cfg.seed)
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_env.py:79]: need to also modify the marl direct env

**Review by Mayankm96 — APPROVED** [review]

File: source/standalone/workflows/rl_games/train.py [hunk 6]
```diff
@@ -76,6 +77,10 @@ def main(env_cfg: ManagerBasedRLEnvCfg | DirectRLEnvCfg | DirectMARLEnvCfg, agen
     env_cfg.scene.num_envs = args_cli.num_envs if args_cli.num_envs is not None else env_cfg.scene.num_envs
     env_cfg.sim.device = args_cli.device if args_cli.device is not None else env_cfg.sim.device
 
+    # randomly sample a seed if seed = -1
+    if args_cli.seed == -1:
+        args_cli.seed = random.randint(0, 10000)
```
Comment (Mayankm96) [inline source/standalone/workflows/rl_games/train.py:82]: Hmm what if the agent_cfg is -1? Should we handle that? Or it won't happen? 😅
